jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/migration/MigrationManager.java

```diff
@@ -395,17 +395,27 @@ private void updateNodeInstances(NodeInstanceContainer nodeInstanceContainer, Ma
             if (newNodeId == null) {
                 newNodeId = oldNodeId;
             }
+            
             Node upgradedNode = findNodeByUniqueId(newNodeId, nodeContainer);
             if (upgradedNode == null) {
+                Boolean isHidden = (Boolean) ((NodeImpl) ((org.jbpm.workflow.instance.NodeInstance) nodeInstance).getNode()).getMetaData().get("hidden");
+                if (isHidden != null && isHidden.booleanValue()) {
+                    // if node is hidden it cannot be found nor migrated, skip it
+                    continue;
+                }
                 try {
                     upgradedNodeId = Long.parseLong(newNodeId);
+                    if (findNodeById(upgradedNodeId, nodeContainer) == null) {
+                        report.addEntry(Type.ERROR, "Node with id " + newNodeId + " was not found in new process definition"); 
+                    }
```

Comment:
So this scenario happens only if we provide mappings of UUID/UniqueId (of an old process) to nodeId (of a new process), right? Then the parseLong is successful and it will try to run findNodeById() I guess. So in the test you added, it will fail to parse Long, since it has a format of UUID, right? Just so I understand in which scenarios this is actually run.

Reply:
correct, if the UUID was not found we try t use it as node id and then look up to make sure such node exists. That is sort of last resort type of validation which in reality (after 6.x) should not happen.

Reply:
Thanks. One more thing I have just realized - shouldn't we assign the node to upgradedNode variable if it is found - so the next IF statement can update logs? (Right now in this case the `upgradedNode != null` is false.)

Reply:
since that is a numeric value that would indicate it's already a node id that is the same as before so there is no need to update anything.

Reply:
Hmm, so we don't need to update nodeType, node name and nodeId in the logs? What if the node name of the new node is different for example?

Reply:
correct, it should not be updated it's considered to be the same.